integration-tests/subscriptions.test.ts

```diff
@@ -86,6 +86,90 @@ if (SUBSCRIPTIONS_ENABLED === 'true') {
                 expect(notifications).toEqual([]);
             });
         }
+
+        test('end to end test with id notifications', async () => {
+            const uuid = v4();
+            const subResource = randomSubscription(uuid);
+            // 1. Create a Subscription.
+            const postSubscriptionResult = await client.post('Subscription', subResource);
+            expect(postSubscriptionResult.status).toBe(201);
+            // 2. Create/Update a resource that matches the subscription.
+            const postPatientResult = await client.post('Patient', resourceThatMatchesSubscription);
+            expect(postPatientResult.status).toEqual(201);
+            // 3. Verify that notifications are received
+            // give 2 minutes for notification to be placed in ddb table
+            await new Promise((r) => setTimeout(r, 120_000));
+            // make sure notification was receieved
+            let notifications = await subscriptionsHelper.getNotifications(
+                `/${uuid}/Patient/${postPatientResult.data.id}`,
+            );
+            expect(notifications).not.toEqual([]);
+            expect(notifications[0].httpMethod).toEqual('PUT');
+            expect(notifications[0].body).toBeNull();
+            // 4. Delete the Subscription
+            const deleteSubscriptionResult = await client.delete(`Subscription/${postSubscriptionResult.data.id}`);
+            expect(deleteSubscriptionResult.status).toEqual(200);
+            await new Promise((r) => setTimeout(r, 120_000));
+            // 5. Create/Update a resource that matches the subscription.
+            // test update:
+            const updatePatientResult = await client.put(`Patient/${postPatientResult.data.id}`, {
+                ...resourceThatMatchesSubscription,
+                id: postPatientResult.data.id,
+            });
+            expect(updatePatientResult.status).toEqual(200);
+            // test create:
+            const postAnotherPatientResult = await client.post('Patient', resourceThatMatchesSubscription);
+            expect(postAnotherPatientResult.status).toEqual(201);
+            // 6. Verify that notifications are no longer being sent
+            await new Promise((r) => setTimeout(r, 120_000));
+            notifications = await subscriptionsHelper.getNotifications(`/${uuid}/Patient/${postPatientResult.data.id}`);
+            // we still have the one notification from earlier in the test, but no more
+            expect(notifications.length).toEqual(1);
```

Comment:
I think that we also need to lookup notifications for `postAnotherPatientResult.data.id` and make sure that there are none.